pennylane/operation.py

```diff
@@ -713,6 +715,44 @@ def __rmatmul__(self, other):
 
     __imatmul__ = __matmul__
 
+    @property
+    def eigvals(self):
+        """Return the eigenvalues of the specified tensor product observable by
+        using pre-stored eigenvalues for standard observables where possible.
```

Comment:
For the first line of the docstring, usually good to keep it as short and concise as possible:
```rest
Return the eigenvalues of the specified tensor product observable.
```

The implementation details (i.e., using pre-stored eigenvalues) can then go in the middle of the docstring.